pkg/util/scrubber/scrubber.go

```diff
@@ -213,11 +214,44 @@ func (c *Scrubber) scrub(data []byte, replacers []Replacer) []byte {
 		}
 		if len(repl.Hints) == 0 || containsHint {
 			if repl.ReplFunc != nil {
-				data = repl.Regex.ReplaceAllFunc(data, repl.ReplFunc)
+				data = repl.Regex.ReplaceAllFunc(data, func(match []byte) []byte {
+					if containsValidENC(match) {
+						return match // Don't replace if contains valid ENC[]
+					}
+					return repl.ReplFunc(match)
+				})
 			} else {
-				data = repl.Regex.ReplaceAll(data, repl.Repl)
+				data = repl.Regex.ReplaceAllFunc(data, func(match []byte) []byte {
+					if containsValidENC(match) {
+						return match // Don't replace if contains valid ENC[]
+					}
+					return repl.Regex.ReplaceAll(match, repl.Repl)
+				})
```

Comment:
Let's invert the logic to avoid duplicating too much code:

```golang
data = repl.Regex.ReplaceAllFunc(data, func(match []byte) []byte {
	if containsValidENC(match) {
		return match // Don't replace if contains valid ENC[]
	}
	if repl.ReplFunc != nil {
		return repl.ReplFunc(match)
	} else {
		return repl.Regex.ReplaceAll(match, repl.Repl)
	}
})
```